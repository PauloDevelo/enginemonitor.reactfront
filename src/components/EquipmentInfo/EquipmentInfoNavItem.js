import React from 'react';
import { NavItem, NavLink } from 'reactstrap';
import PropTypes from 'prop-types';
import classnames from 'classnames';

export default function EquipmentInfoNavItem({equipment, active, onClick}){
    return(
        <NavItem>
            <NavLink className={classnames({ active: active })} onClick={() => { 
                if(onClick)
                    onClick();
             }}>
                {equipment.name}
            </NavLink>
        </NavItem>
    );
}

EquipmentInfoNavItem.propTypes = {
    equipment: PropTypes.object.isRequired,
    active: PropTypes.bool,
    onClick: PropTypes.func
};