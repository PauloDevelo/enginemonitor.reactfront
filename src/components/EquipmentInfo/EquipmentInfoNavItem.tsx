/* eslint-disable no-unused-vars */
/* eslint-disable react/require-default-props */

import React, { useCallback } from 'react';
import { NavItem, NavLink } from 'reactstrap';
import classnames from 'classnames';
// eslint-disable-next-line no-unused-vars
import { EquipmentModel } from '../../types/Types';

type Props = {
    equipment: EquipmentModel,
    active: boolean,
    setCurrentEquipment?: (equipment: EquipmentModel) => void
};

function EquipmentInfoNavItem({ equipment, active, setCurrentEquipment }: Props) {
  const setCurrentEquipmentCallBack = useCallback(() => {
    if (setCurrentEquipment) { setCurrentEquipment(equipment); }
  }, [equipment, setCurrentEquipment]);

  return (
    <NavItem>
      <NavLink className={classnames({ active })} onClick={setCurrentEquipmentCallBack}>
        {equipment.name}
      </NavLink>
    </NavItem>
  );
}

export default React.memo(EquipmentInfoNavItem);
