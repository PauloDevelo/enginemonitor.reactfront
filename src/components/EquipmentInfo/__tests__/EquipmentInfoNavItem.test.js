
import { mount } from 'enzyme';
import ignoredMessages from '../../../testHelpers/MockConsole';

import React from 'react';

import EquipmentInfoNavItem from '../EquipmentInfoNavItem';

describe("EquipmentInfoNavItem", () => {

    const equipment = {
        _uiId: '1234',
        name: 'engine',
        brand: 'nanni',
        model: 'N3.30',
        age: 2563,
        installation: new Date('2011-02-22T16:00:00.000Z')
    }

    beforeAll(() => {
        ignoredMessages.length = 0;
    });

    beforeEach(() => {
        
    });

    afterEach(() => {

    });

    it("Should render an tab nav item not active", () => {
        // Arrange
        const setCurrentEquipment = jest.fn();

        // Act
        const equipmentInfoNavItem = mount(<EquipmentInfoNavItem equipment={equipment} active={false} setCurrentEquipment={setCurrentEquipment}/>);
        
        // Assert
        expect(setCurrentEquipment).toHaveBeenCalledTimes(0);
        expect(equipmentInfoNavItem.find('NavLink').text()).toBe(equipment.name);
        expect(equipmentInfoNavItem).toMatchSnapshot();
    });

    it("Should render an tab nav item active", () => {
        // Arrange
        const setCurrentEquipment = jest.fn();

        // Act
        const equipmentInfoNavItem = mount(<EquipmentInfoNavItem equipment={equipment} active={true} setCurrentEquipment={setCurrentEquipment}/>);
        
        // Assert
        expect(setCurrentEquipment).toHaveBeenCalledTimes(0);
        expect(equipmentInfoNavItem.find('NavLink.active').text()).toBe(equipment.name);
        expect(equipmentInfoNavItem).toMatchSnapshot();
    });

    it("Should call setCurrentEquipment after clicking on it", () => {
        // Arrange
        const setCurrentEquipment = jest.fn();
        const equipmentInfoNavItem = mount(<EquipmentInfoNavItem equipment={equipment} active={false} setCurrentEquipment={setCurrentEquipment}/>);

        // Act
        equipmentInfoNavItem.find('NavLink').simulate('click');
        
        // Assert
        expect(setCurrentEquipment).toHaveBeenCalledTimes(1);
    });
});