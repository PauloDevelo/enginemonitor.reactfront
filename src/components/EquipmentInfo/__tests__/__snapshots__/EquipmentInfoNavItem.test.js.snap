// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EquipmentInfoNavItem Should render an tab nav item active 1`] = `
<Memo(EquipmentInfoNavItem)
  active={true}
  equipment={
    Object {
      "_uiId": "1234",
      "age": 2563,
      "brand": "nanni",
      "installation": 2011-02-22T16:00:00.000Z,
      "model": "N3.30",
      "name": "engine",
    }
  }
  setCurrentEquipment={[MockFunction]}
>
  <NavItem
    tag="li"
  >
    <li
      className="nav-item"
    >
      <NavLink
        className="active"
        onClick={[Function]}
        tag="a"
      >
        <a
          className="active nav-link"
          onClick={[Function]}
        >
          engine
        </a>
      </NavLink>
    </li>
  </NavItem>
</Memo(EquipmentInfoNavItem)>
`;

exports[`EquipmentInfoNavItem Should render an tab nav item not active 1`] = `
<Memo(EquipmentInfoNavItem)
  active={false}
  equipment={
    Object {
      "_uiId": "1234",
      "age": 2563,
      "brand": "nanni",
      "installation": 2011-02-22T16:00:00.000Z,
      "model": "N3.30",
      "name": "engine",
    }
  }
  setCurrentEquipment={[MockFunction]}
>
  <NavItem
    tag="li"
  >
    <li
      className="nav-item"
    >
      <NavLink
        className=""
        onClick={[Function]}
        tag="a"
      >
        <a
          className="nav-link"
          onClick={[Function]}
        >
          engine
        </a>
      </NavLink>
    </li>
  </NavItem>
</Memo(EquipmentInfoNavItem)>
`;
